slamd/discovery/processing/algorithms/experiment_postprocessor.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import pandas as pd
+
+from slamd.discovery.processing.algorithms.plot_generator import PlotGenerator
+
+
+class ExperimentPostprocessor:
+    @classmethod
+    def postprocess(cls, exp):
+        # Construct dataframe for output
+
+        df = exp.orig_data.loc[exp.nolabel_index].copy()
+        # Add the columns with utility and novelty values
+        df['Utility'] = exp.utility.round(6)
+        df['Novelty'] = exp.novelty.round(6)
+
+        for target in exp.target_names:
+            df.loc[exp.nolabel_index, target] = exp.prediction[target].round(6)
+            df[f'Uncertainty ({target})'] = exp.uncertainty[target].round(5)
+
+        df = cls.preprocess_dataframe_for_output_table(df, exp)
```

Comment:
Here we have a preprocess method call within a postprocess method. Maybe we can find a less confusing name instead of "preprocess_dataframe_for_output_table"

Reply:
Yeah, that occured to me as well this morning. Renamed it to "process_dataframe_for_output_table"